### PR TITLE
Allow setting --node-num and --machine-type for GKE deployer

### PR DESCRIPTION
**Pull request by chizhg**
As we discussed offline, allow setting nodes and machineType via --node-num and --machine-type flags for GKE deployer, rather than always using the default value.

/cc @amwat

### CODE REVIEW

File: kubetest2-gke/deployer/deployer.go [hunk 1]
```diff
@@ -185,8 +190,10 @@ func bindFlags(d *deployer) *pflag.FlagSet {
 	flags.StringVar(&d.network, "network", "default", "Cluster network. Defaults to the default network.")
 	flags.StringVar(&d.environment, "environment", "staging", "Container API endpoint to use, one of 'test', 'staging', 'prod', or a custom https:// URL")
 	flags.StringVar(&d.project, "project", "", "Project to deploy to.")
-	flags.StringVar(&d.region, "region", "", "For use with gcloud commands")
-	flags.StringVar(&d.zone, "zone", "", "For use with gcloud commands")
+	flags.StringVar(&d.region, "region", "", "For use with gcloud commands to specify the cluster region.")
+	flags.StringVar(&d.zone, "zone", "", "For use with gcloud commands to specify the cluster zone.")
+	flags.IntVar(&d.nodes, "node-num", defaultNodePool.Nodes, "For use with gcloud commands to specify the number of nodes for the cluster.")
```
Comment (amwat) [inline kubetest2-gke/deployer/deployer.go:195]: let's just name it `num-nodes` for consistency.

Comment (chizhg) [reply kubetest2-gke/deployer/deployer.go]: Done.

File: kubetest2-gke/deployer/deployer.go [hunk 2]
```diff
@@ -139,6 +141,9 @@ func (d *deployer) verifyUpFlags() error {
 	if _, err := d.location(); err != nil {
 		return err
 	}
+	if d.nodes <= 0 {
+		return fmt.Errorf("--node-num must be larger than 0")
```
Comment (amwat) [inline kubetest2-gke/deployer/deployer.go:145]: here too

Comment (chizhg) [reply kubetest2-gke/deployer/deployer.go]: Sorry missed that, done.

**Pull request merged**